# Basic setup for fetching data from the facebook api

### Remember to add an active access token below and adjust the parameters and fields if necessary

In [13]:
# must be specified to be able to use methods of package
import sys
sys.path.append('../src/') 
# import packages
import pandas as pd



In [3]:
# # insert your access token here (expires every 2 hours)
# access_token = "EAADqFZC76BDMBAO2QwwFelyBTOYZB0Nig3jw6xpHVBo8OTPemZBNWVNZCxQPjFf6UdRNDNSfxsxdUkp6Of3W0vNMzyq0B12hVyv3uQkVcmQoomRL1bm02FRJt2m1N8tiA54T6fN8gPLOoV6AvtTXhbQ5nE6xZAej2c09s3BmPEv5N0kBEZBa76ArjXyWcZByaNRxE4cCmo3GdZBZAnYa0ytEZA8ZCPq5PiZAgknO5U6UcFZAR8x1MPnywIZAufQSg6ZCbUXAfYZD"
# # get_new_access_token: https://developers.facebook.com/tools/explorer/


# #https://www.facebook.com/ads/library/report
# # "../data/generated_dataset.txt",

# # PARAMETERS NECESSARY
# ad_type = "POLITICAL_AND_ISSUE_ADS"
# ad_reached_countries = ['US'] # Facebook delivered the ads in these countries. Provided as ISO country codes.
# search_terms = "Joe Biden" # The terms to search for in your query. We treat a blank space as a logical AND and search for both terms and no other operators. The limit of your string is 100 characters or less.
# limit = 1700
# start_date = "2020-08-00"
# end_date = "2020-11-03"

# # FIELDS to specify your results
# fields = ["id", "page_name", "ad_creative_body", "impressions", "delivery_by_region", "demographic_distribution", "ad_delivery_start_time", "bylines", "spend"]

# # see here for more parameters and fields: https://www.facebook.com/ads/library/api/?source=archive-landing-page 

# input_url = f"https://graph.facebook.com/v12.0/ads_archive?search_terms={search_terms}&ad_type={ad_type}&ad_reached_countries={ad_reached_countries}&fields={fields}&limit={limit}&ad_delivery_date_min={start_date}&ad_delivery_date_max={end_date}"

# api = API_request()

# response = api.get_json_response(input_url, access_token)

# data = pd.DataFrame(response["data"])


In [ ]:
# import time 

# #https://developers.facebook.com/docs/graph-api/results

# iterations = 10
# sleep_time_sec = 10

# counter = 0

# for i in range(10):

#     paging = response["paging"]["next"]
#     next_response = get_json_response(paging, access_token)

#     new_page = pd.DataFrame(next_response['data'])
#     data = data.append(new_page, ignore_index=True)
#     time.sleep(10)
#     counter+=1
#     print(counter)


# Looking at the data-set with all the politicians

## Making aggregate methods from their target groups (Demographic DATA)



In [3]:
#Code to read from our full dataset:
preprocess = Preprocessor()
df = preprocess.file_to_df('/home/gustavgyrst/Desktop/research_project/political-ad-api/data/all_politicians_aggregated.txt')

In [4]:
df['demographic_distribution'].iloc[0]
df.dropna(inplace=True) #

In [5]:
# data.drop(['age_distribution', 'gender'], axis=1)

#data['age_distribution'] = data['demographic_distribution'].apply(lambda x: get_age_distribution(x, all_ages=True))

demographic_data = df[['page_name', 'demographic_distribution']].reset_index().drop(columns='index')

# Finding the Missing Page_names

In [15]:
df_missing = pd.read_csv('../src/data_sets/missing_pid.csv')
df_all_ad_pages = pd.read_csv('../data/FacebookAdLibraryReport_2021-10-15_US_lifelong_advertisers.csv')



/home/gugy/anaconda3/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (3) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [23]:
pol = pd.read_csv('../src/data_sets/US_legislators_page_ids_2021.csv')
import time

In [33]:
for name in pol.full_name.to_list():
    res = df_all_ad_pages[df_all_ad_pages['Page Name'].str.contains(name, na=False)][['Page ID', 'Page Name', 'Amount Spent (USD)']].head(5)
    print(res)
    

          Page ID      Page Name Amount Spent (USD)
763    6266829799  Sherrod Brown             508269
7674   6266829799  Sherrod Brown              35660
43976  6266829799  Sherrod Brown               2056
         Page ID       Page Name Amount Spent (USD)
1395  9351652533  Maria Cantwell             284524
Empty DataFrame
Columns: [Page ID, Page Name, Amount Spent (USD)]
Index: []
Empty DataFrame
Columns: [Page ID, Page Name, Amount Spent (USD)]
Index: []
Empty DataFrame
Columns: [Page ID, Page Name, Amount Spent (USD)]
Index: []
         Page ID         Page Name Amount Spent (USD)
7304  6314129300  Dianne Feinstein              38132
          Page ID      Page Name Amount Spent (USD)
174    7606381190  Amy Klobuchar            2014000
519    7606381190  Amy Klobuchar             749441
4757   7606381190  Amy Klobuchar              67698
32121  7606381190  Amy Klobuchar               3734
43631  7606381190  Amy Klobuchar               2087
Empty DataFrame
Columns: [Page ID, Page 

In [17]:
df_all_ad_pages[df_all_ad_pages['Page Name'].str.contains('French Hill', na=False)].head(50)

Page ID                 Page Name  \
2146    1374601146109275  French Hill for Arkansas   
12767    727280547367245   Congressman French Hill   
23844   1374601146109275  French Hill for Arkansas   
25976    727280547367245   Congressman French Hill   
31694   1374601146109275  French Hill for Arkansas   
75033    727280547367245   Congressman French Hill   
188507   727280547367245   Congressman French Hill   

                                Disclaimer Amount Spent (USD)  \
2146              French Hill for Arkansas             182343   
12767              Congressman French Hill              17270   
23844   These ads ran without a disclaimer               6318   
25976    Office of Congressman French Hill               5437   
31694             FRENCH HILL FOR ARKANSAS               3828   
75033           Representative French Hill                654   
188507  These ads ran without a disclaimer               ≤100   

        Number of Ads in Library  
2146                         595  
12767                         78  
23844                          6  
25976                         85  
31694                         27  
75033                          5  
188507                         1

In [16]:
df_all_ad_pages[df_all_ad_pages['Page Name'].str.contains('Vicente Gonzalez', na=False)].head(50)

Page ID                               Page Name  \
13616    206690159663661  U.S. Congressman Vicente Gonzalez 2022   
30729    206690159663661  U.S. Congressman Vicente Gonzalez 2022   
48586   1838012813083658            Congressman Vicente Gonzalez   
60721   1838012813083658            Congressman Vicente Gonzalez   
90177   1838012813083658            Congressman Vicente Gonzalez   
168447   206690159663661  U.S. Congressman Vicente Gonzalez 2022   
223284   206690159663661  U.S. Congressman Vicente Gonzalez 2022   

                                          Disclaimer Amount Spent (USD)  \
13616                  VICENTE GONZALEZ FOR CONGRESS              15688   
30729        Vicente Gonzalez for Congress Committee               4030   
48586                   Congressman Vicente Gonzalez               1696   
60721             These ads ran without a disclaimer               1050   
90177                   Congressman Vicente Gonzalez                420   
168447  Vicente Gonzalez for U.S. Congress Committee               ≤100   
223284            These ads ran without a disclaimer               ≤100   

        Number of Ads in Library  
13616                        292  
30729                        173  
48586                          4  
60721                          4  
90177                          2  
168447                         9  
223284                         6

In [6]:
df_missing[df_missing.page_id == 'no match'].count()

Unnamed: 0         166
first_name         166
last_name          166
full_name          166
type               166
state              166
district           136
party              166
facebook           132
page_name           67
identifier         166
page_id            166
already_fetched    166
dtype: int64

In [7]:
#Filtering with two values with a regex AND operation (e.g., first name "Bernie" AND "Sanders" will return all results that contain both Bernie and Sanders)
def get_page_id(data: pd.DataFrame, first_name: str, last_name:str):
    match = data[data["Page Name"].str.contains(r'^(?=.*'+first_name+')(?=.*'+last_name+')', na=False)]
    if len(match) != 0:
        return match.iloc[0][0]
    else:
        return "no match"


In [7]:
#Testing Regex filtering:
f = 'Bernie'
l = 'Sanders'
match = df_all_ad_pages[df_all_ad_pages["Page Name"].str.contains(r'^(?=.*'+f+')(?=.*'+l+')', na=False)]
match

Page ID                              Page Name  \
26       124955570892789                         Bernie Sanders   
77       124955570892789                         Bernie Sanders   
1443     124955570892789                         Bernie Sanders   
14317   1645816202305254          The People For Bernie Sanders   
44528    107229050703874                Bernie Sanders Fan Club   
...                  ...                                    ...   
352660   799023333537673             Denmark for Bernie Sanders   
368255   427298174139138     Sedgwick County for Bernie Sanders   
372910   142457072843777         Rogers Park For Bernie Sanders   
373310   158942027817875  Filipino Americans for Bernie Sanders   
381564  1672284073041311                      US Bernie Sanders   

                                Disclaimer Amount Spent (USD)  \
26                             BERNIE 2020            9199592   
77                            Bernie 2020.            4132342   
1443             Friends of Bernie Sanders             274572   
14317            Paid for by Be a Hero PAC              14463   
44528                        Imran M Jemal               2006   
...                                    ...                ...   
352660  These ads ran without a disclaimer               ≤100   
368255  These ads ran without a disclaimer               ≤100   
372910  These ads ran without a disclaimer               ≤100   
373310  These ads ran without a disclaimer               ≤100   
381564  These ads ran without a disclaimer               ≤100   

        Number of Ads in Library  
26                         23830  
77                         45749  
1443                         825  
14317                        120  
44528                        175  
...                          ...  
352660                         2  
368255                         1  
372910                         1  
373310                         1  
381564                         1  

[86 rows x 5 columns]

In [8]:
#Extracting the data to the df with all the fetched new page_ids
df_missing["page_id"] = df_missing.apply(lambda x: get_page_id(df_all_ad_pages, first_name=x['first_name'], last_name=x['last_name']),axis=1)


In [10]:
print(df_missing[df_missing.page_id == 'no match'].count()[0]) #value of no matches left

66


In [9]:
#Filtering by just one value
def get_page_id1(data: pd.DataFrame, name: str):
    
    match = data[data["Page Name"].str.contains(name, na=False)]
    if len(match) != 0:
        return match.iloc[0][0]
    else:
        return "no match"


In [11]:
#Now Cleaning the remaining with only last-name match/search
df_missing['page_id'] = df_missing.apply(lambda x: get_page_id1(df_all_ad_pages, name=x['last_name']) if x['page_id'] == 'no match' else x['page_id'], axis=1)

In [12]:
print(df_missing[df_missing.page_id == 'no match'].count()[0]) #value of no matches left
len(df_missing) #total length

7


181

# Quality Control and Cleaning Faulty Values

In [19]:
def quality_control_page_id(df, df_transparency_report):
    df_without_no_match = df[df.page_id != 'no match']
    for id in df_without_no_match.page_id:
        df_temp = df_transparency_report[df_transparency_report['Page ID'] == id]
        print("Page name transparency report: " + df_temp['Page Name'].unique())
        df_temp2 = df_without_no_match[df_without_no_match['page_id'] == id]
        print("Missing list: df_temp2: "  + df_temp2['full_name'])



In [ ]:
#Template for the filtering method

#find the replacement value in the transparency report - looked up with parsing the correct facebook page name
replacement = df_all_ad_pages[df_all_ad_pages['Page Name'] == 'Bob Menendez']['Page ID'].iloc[0] #Have to add the iloc[0] to only get the specific value and the whole data series
row_index = df_missing[df_missing['full_name'] == 'Robert Menendez'].page_id.index[0]

type(replacement)
replacement

#column number for page_id
df_missing.columns[11]
df_missing.iat[row_index, 11] = replacement

row_index

4

In [11]:
def replace_false_pages(dict, df_missing, df_transparency_report):
    for key, value in dict.items():
        print(key, value)
        row_index = df_missing[df_missing['full_name'] == key].page_id.index[0]    
        print(row_index)
        if value == 'no ad account':
            replacement = 'no ad account'
        else:
            replacement = df_transparency_report[df_transparency_report['Page Name'] == value]['Page ID'].iloc[0] #Have to add the iloc[0] to only get the specific value and the whole data series
        
        df_missing.iat[row_index, 11] = replacement



Henry C. "Hank" Johnson, Jr.Congressman Hank Johnson
John H. RutherfordNo Ad Account
Robert MenendezBob Menendez


In [34]:
#Finding the faulty fetches:
dict_of_faulty_pid = {
        'Henry C. "Hank" Johnson, Jr.': 'Congressman Hank Johnson',
        'John H. Rutherford':'no ad account',
        'Robert Menendez': 'Bob Menendez',
        'Benjamin L. Cardin': 'no ad account',
        'H. Morgan Griffith': 'Morgan Griffith for Congress',
        'Eddie Bernice Johnson': 'no ad account',
        'Gregory W. Meeks': 'no ad account',
        'Jerry Moran':'no ad account',
        'Harold Rogers':'Hal Rogers',
        'Donald M. Payne, Jr.': 'no ad account',
        'Robin L. Kelly': 'no ad account',
        'Donald S. Beyer, Jr.': 'no ad account',
        'J. Luis Correa':'Rep. Lou Correa',
        'A. Drew Ferguson IV':'Congressman Drew Ferguson',
        'W. Gregory Steube':'Congressman Greg Steube',
        'C. Scott Franklin':'Congressman Scott Franklin',
        'Bradley Scott Schneider':'Congressman Brad Schneider',
        'Scott DesJarlais': 'no ad account',
        'Tom O’Halleran':'Rep. Tom O\'Halleran',
        'Louie Gohmert': 'no ad account',
        'John Hoeven':'no ad account',
        'Lisa Murkowski':'no ad account',
        'Aumua Amata Coleman Radewagen':'no ad account',
        'Jenniffer González-Colón':'no ad account'
        }


## Documentation of filtering process:

The page_ids for the ad campaign pages of the politicians are not accessible anywhere publically. Therefore, we have used the transparency report of all ad pages to fetch the page ids of the US Congress members. This is done in three steps. First we filter by "Full_name". This was highly efficient as most politicians used their full name for their page name. 

The second phase of filtering then occuring by a regex statement with first and last name. This gets us down only missing 66 politicians.

For the last 66 polticians we use the lastname alone for the filtering. 


There are several cases when we have tried to filter out pages that it resulted in a false data_set, some examples are presented below.
I had to go through full set of 180 manually to filter these cases out and clean the data. This was gone by cross checking and finding their facebook page, then trying to find their facebook account in the transparency report (both in the dataframe version and the website interactive page)

### Examples: 

["Page name transparency report: Mark Amodei Isn't Fighting for Us"]
88    Missing list: df_temp2: Mark E. Amodei

['Page name transparency report: Sharon Whitehurst-Payne for San Diego School Board']
90    Missing list: df_temp2: Donald M. Payne, Jr.

['Page name transparency report: Captain Mark Kelly']
102    Missing list: df_temp2: Robin L. Kelly

While Robin Kelly has a facebook account with many followers, she does not have an ad account.
https://www.facebook.com/reprobinkelly

['Page name transparency report: Jay Schneiderman for Southampton Town Supervisor']
123    Missing list: df_temp2: Bradley Scott Schneider


['Page name transparency report: Help Bring Justice To Robert D. Correale, Esq., his former law firm, Maynard & Truland']
127    Missing list: df_temp2: J. Luis Correa

These have been cleaned out by going through the dataset manually.

### Could be good to add multiple pages for some of them? Or at least we have mention this in our limitations
additional - extra pages for politicians:
Drew Ferguson
Greg Steube

### all polticians have facebook pages not all have ad accounts 

--> a lot have multiple pages (both 'senator bob connison' and 'bob connison')


In [39]:
replace_false_pages(dict_of_faulty_pid, df_missing, df_all_ad_pages)

Henry C. "Hank" Johnson, Jr. Congressman Hank Johnson
41
John H. Rutherford no ad account
129
Robert Menendez Bob Menendez
4
Benjamin L. Cardin no ad account
1
H. Morgan Griffith Morgan Griffith for Congress
34
Eddie Bernice Johnson no ad account
40
Gregory W. Meeks no ad account
54
Jerry Moran no ad account
55
Harold Rogers Hal Rogers
66
Donald M. Payne, Jr. no ad account
90
Robin L. Kelly no ad account
102
Donald S. Beyer, Jr. no ad account
117
J. Luis Correa Rep. Lou Correa
127
A. Drew Ferguson IV Congressman Drew Ferguson
134
W. Gregory Steube Congressman Greg Steube
142
C. Scott Franklin Congressman Scott Franklin
166
Bradley Scott Schneider Congressman Brad Schneider
123
Scott DesJarlais no ad account
27
Tom O’Halleran Rep. Tom O'Halleran
124
Louie Gohmert no ad account
32
John Hoeven no ad account
38
Lisa Murkowski no ad account
56
Aumua Amata Coleman Radewagen no ad account
121
Jenniffer González-Colón no ad account
138


In [24]:
df_all_ad_pages[df_all_ad_pages['Page Name'].str.contains('Maria Cantwell', na=False)].head(20)

Page ID       Page Name        Disclaimer Amount Spent (USD)  \
1395  9351652533  Maria Cantwell  Friends of Maria             284524   

      Number of Ads in Library  
1395                       102

In [36]:
df_missing[df_missing.page_id.str.contains("no match", na=False)]

Empty DataFrame
Columns: [Unnamed: 0, first_name, last_name, full_name, type, state, district, party, facebook, page_name, identifier, page_id, already_fetched]
Index: []

In [56]:
#df_missing.to_csv(path_or_buf='/home/gustavgyrst/Desktop/Research Project/political-ad-api/src/data_sets/found_missing_pid.csv')


Page ID                 Page Name  \
2146    1374601146109275  French Hill for Arkansas   
12767    727280547367245   Congressman French Hill   
23844   1374601146109275  French Hill for Arkansas   
25976    727280547367245   Congressman French Hill   
31694   1374601146109275  French Hill for Arkansas   
75033    727280547367245   Congressman French Hill   
188507   727280547367245   Congressman French Hill   

                                Disclaimer Amount Spent (USD)  \
2146              French Hill for Arkansas             182343   
12767              Congressman French Hill              17270   
23844   These ads ran without a disclaimer               6318   
25976    Office of Congressman French Hill               5437   
31694             FRENCH HILL FOR ARKANSAS               3828   
75033           Representative French Hill                654   
188507  These ads ran without a disclaimer               ≤100   

        Number of Ads in Library  
2146                         595  
12767                         78  
23844                          6  
25976                         85  
31694                         27  
75033                          5  
188507                         1

In [76]:
df_all_ad_pages[df_all_ad_pages['Page Name'].str.contains('Claudia Tenney', na=False)].head(50)

Page ID                                          Page Name  \
3306   2220944231249057                        Claudia Tenney Wrong for NY   
4675    464292670410321                        Claudia Tenney for Congress   
7313    100910491745994  Claudia Tenney: In the pocket of corporate spe...   
11788   183020072173458                                Rep. Claudia Tenney   
28843   464292670410321                        Claudia Tenney for Congress   
53500  1961305130853062               Claudia Tenney's Healthcare Disaster   
64551  1961305130853062               Claudia Tenney's Healthcare Disaster   

                                              Disclaimer Amount Spent (USD)  \
3306   DCCC. 430 South Capitol St, S.E. Washington, D...             107390   
4675                                 TENNEY FOR CONGRESS              69277   
7313                                                DCCC              38064   
11788             Office of Congresswoman Claudia Tenney              19371   
28843                        Claudia Tenney for Congress               4510   
53500                 These ads ran without a disclaimer               1388   
64551                                 House Majority PAC                926   

       Number of Ads in Library  
3306                         50  
4675                        184  
7313                         38  
11788                        57  
28843                        32  
53500                        26  
64551                        31

In [5]:
df_pol = pd.read_csv('/home/gustavgyrst/Desktop/Research Project/political-ad-api/src/data_sets/legislators_fb_info.csv')


In [7]:
df_for_quality_control = df_pol[df_pol.already_fetched == 'done']

df_for_quality_control[df_for_quality_control.full_name == 'John Kennedy']

Unnamed: 0 first_name last_name     full_name type state  district  \
322         322       John   Kennedy  John Kennedy  sen    LA       NaN   

          party              facebook page_name  identifier          page_id  \
322  Republican  JohnKennedyLouisiana       NaN           0  491347857894708   

    already_fetched  
322            done

In [8]:
df_without_no_match = df_for_quality_control[df_for_quality_control.page_id != 'no match']
for id in df_without_no_match.page_id:
    id = int(id)
    df_temp = df_all_ad_pages[df_all_ad_pages['Page ID'] == id]
    print("Page name transparency report: " + df_temp['Page Name'].unique())
    df_temp2 = df_without_no_match[df_without_no_match['page_id'] == str(id)]
    print("Missing list: df_temp2: "  + df_temp2['full_name'])



['Page name transparency report: Sherrod Brown']
0    Missing list: df_temp2: Sherrod Brown
Name: full_name, dtype: object
['Page name transparency report: Dianne Feinstein']
5    Missing list: df_temp2: Dianne Feinstein
Name: full_name, dtype: object
['Page name transparency report: Amy Klobuchar']
6    Missing list: df_temp2: Amy Klobuchar
Name: full_name, dtype: object
['Page name transparency report: Debbie Stabenow']
9    Missing list: df_temp2: Debbie Stabenow
Name: full_name, dtype: object
['Page name transparency report: Jon Tester']
10    Missing list: df_temp2: Jon Tester
Name: full_name, dtype: object
['Page name transparency report: Sheldon Whitehouse']
11    Missing list: df_temp2: Sheldon Whitehouse
Name: full_name, dtype: object
['Page name transparency report: John Barrasso']
12    Missing list: df_temp2: John Barrasso
Name: full_name, dtype: object
['Page name transparency report: John Cornyn']
15    Missing list: df_temp2: John Cornyn
Name: full_name, dtype: object
['

In [ ]:

#Cleaning the 350 some
dict_faulty = {
    'Doug Lamborn': 'no ad account',
    'Richard Hudson': 'Richard Hudson',
    'Glenn Thompson': 'no ad account', 
    'Maxine Waters': 'no ad account',
    'J. Hill': 'French Hill for Arkansas',
    'Thom Tillis': 'Thom Tillis',
    'John Kennedy': 'John Neely Kennedy',
    'Claudia Tenney':'Rep. Claudia Tenney'
}



In [78]:
df_for_quality_control[df_for_quality_control.full_name == 'Claudia Tenney']


Unnamed: 0 first_name last_name       full_name type state  district  \
533         533    Claudia    Tenney  Claudia Tenney  rep    NY      22.0   

          party facebook page_name  identifier           page_id  \
533  Republican      NaN       NaN           0  2220944231249057   

    already_fetched  
533            done

In [12]:
replace_false_pages(dict_faulty, df_pol, df_transparency_report=df_all_ad_pages)


Doug Lamborn no ad account
113
Richard Hudson Richard Hudson
248
Glenn Thompson no ad account
195
Maxine Waters no ad account
205
J. Hill French Hill for Arkansas
279
Thom Tillis Thom Tillis
314
John Kennedy John Neely Kennedy
322
Claudia Tenney Rep. Claudia Tenney
533


In [23]:
df_pol.iat[533,11]
df_pol['full_name']

if 'Thom Tillis' in dict_faulty.keys():
    print(True)


True


In [28]:
df_pol['cleaning_update'] = df_pol.apply(lambda x: "modified" if x['full_name'] in dict_faulty.keys() else "", axis=1)

In [30]:
#df_pol.to_csv('congress_fbNamesV2.csv')

In [32]:
df_pol_mod = df_pol[df_pol.cleaning_update == 'modified']

df_pol_mod.to_csv('cleaning_update.csv')

['Page name transparency report: Friends of Glenn Thompson - Team GT']
195    Missing list: df_temp2: Glenn Thompson

['Page name transparency report: Latinos For Congresswoman Maxine Waters']
205    Missing list: df_temp2: Maxine Waters


['Page name transparency report: Friends of Derek Kilmer']
270    Missing list: df_temp2: Derek Kilmer



['Page name transparency report: Detroit Caring Dentistry directed by Dr. David J. Hill DDS']
279    Missing list: df_temp2: J. Hill

quite a few have two ad accounts as well. Both a general account and one specific to a campaign "French Hill for Arkansas" and "Congressman French Hill"


'Corrupt Thom Tillis', 'Defeat Thom Tillis'



In [ ]:
# #df.loc[df['B'].isin(['one','three'])])

# # Two ways to attack this filtering of the data:
# # data.loc[data['page_name'].isin(['Boost the News'])]
# # data.loc[data['page_name'] == 'Boost the News']

# # data['page_name'].str.contains('Boost', 'Climate').any()

# # data['page_name'].filter(regex='Boost')

# # @USCongressmanFilemonVela etc.
# set_of_pol_pgnames = {'USCongressmanFilemonVela', 'reppaultonko', 'SenatorThomTillis', 'DonaldNorcrossNJ', '153423912663', 'repalgreen', 'RepDonBeyer', 'RepRogerWilliams', 'CongressmanSethMoulton', 'replizcheney', 'RepRoybalAllard', 'mitchmcconnell', 'SenatorRandPaul', 'RepSteveScalise', 'RepMoBrooks', 'CongressmanAlexMooney', 'RepJackBergman', 'RepWesterman', 'RepMcEachin', 'CongresswomanAnnieKuster', 'RepFrankLucas', 'CongressmanTedDeutch', 'GwenSMoore', 'RepTedLieu', 'RepGregMurphy', 'senatordebfischer', 'RepBobGibbs', 'AndyHarrisMD', 'WhipHoyer', 'CongressmanMichaelF.Q.SanNicolas', 'congressmangkbutterfield', 'SenatorTimScott', 'RepJohnCurtis', 'CongressmanNadler', '50375006903', 'RepBarbaraLee', 'jeffmerkley', 'JoeManchinIII', 'RepBost', 'CongressmanBuchanan', 'RepJeffDuncan', 'Congressman.Keating', 'RepSewell', 'RepRickCrawford', 'RepChrisStewart', 'zoelofgren', 'CongressmanWarrenDavidson', 'drnealdunnfl2', 'RepTomSuozzi', 'SenatorLankford', 'RepCheri', 'RepKinzinger', 'marshablackburn', 'SenBrianSchatz', 'RepDonBacon', 'SenatorHassan', 'senshelley', 'senatorcantwell', 'BlaineLuetkemeyer', '113303673339', 'RepJohnLarson', 'senatortester', 'RepRouzer', 'RobertAderholt', 'reptrey', 'jameseclyburn', 'RepRichHudson', 'RickScottSenOffice', 'SenGaryPeters', 'repschrader', 'RepKweisiMfume', 'CongressmanSteveCohen', 'senatortammybaldwin', 'RepAdamSchiff', 'RepBobbyScott', 'RepEliseStefanik', 'senatorelizabethwarren', 'CongressmanMattCartwright', 'RepAmiBera', 'CongressmanKevinMcCarthy', 'RepAnthonyBrown', 'repgaramendi', 'CongressmanDavidCicilline', 'RepJimBanks', 'JodeyArrington', 'repgracemeng', 'bill.posey15', 'senatorchriscoons', 'RepTedBudd', 'RepBrianHiggins', 'RepMarkTakano', 'CongressmanBennieGThompson', 'emanuelcleaverii', 'RepMarkDeSaulnier', 'RepHuffman', 'SenatorTomCotton', 'repscottperry', 'RepresentativeValDemings', 'senatorjimrisch', 'ScottDesJarlaisTN04', 'RepLarryBucshon', 'repchuck', 'RepWalberg', 'senbennetco', 'RepLoisFrankel', 'reptomrice', 'CongressmanClayHiggins', 'SenatorJohnHoeven', 'RepBonnie', 'RepHuizenga', 'CongressmanMikeDRogers', 'RepJoyceBeatty', 'SenatorSasse', 'CongressmanMarcVeasey', 'DianaDeGette', 'chrismurphyct', 'gregorymeeksny05', 'CongressmanDKDavis', 'CongresswomanRosaDeLauro', 'senatorbencardin', 'SenatorMikeRounds', 'RepAustinScott', 'mcmorrisrodgers', '152569121550', 'RepJuliaBrownley', 'USSenTinaSmith', 'repmarkpocan', 'timryan', 'herrerabeutler', 'sen.johncornyn', 'repstephenlynch', 'RepJoshG', 'RepJuanVargas', 'CongressmanGuthrie', 'repfrankpallone', '81058818750', 'tomcarper', 'repgosar', 'senjoniernst', 'CongressmanBillFoster', 'repcardenas', 'SenatorKaine', 'RepJimMcGovern', 'RepConorLamb', 'RepAdamSmith', 'RepDavidKustoff', 'RepDebbieLesko', 'RepKathleenRice', 'johnbarrasso', 'SenJackReed', 'RepJimmyPanetta', 'RepSteveChabot', 'reprobinkelly', 'pascrell', 'CongressmanKevinCramer', 'CongressmanGarretGraves', 'JohnKennedyLouisiana', 'senatorfeinstein', 'RepMullin', 'RepRubenGallego', 'RepScottPeters', 'joecourtney', 'RepSmucker', 'TomColeOK04', 'CongresswomanLindaSanchez', 'CongressmanDougLamborn', 'RepMikeJohnson', 'SenatorRichardBurr', 'repbrianmast', 'repjimjordan', 'RepMikeTurner', 'DonaldPayneJr', 'SenatorPatrickLeahy', 'RepTomReed', 'CongressmanBoyle', 'SenatorCortezMasto', 'ChelliePingree', 'RepRoKhanna', 'senatormikelee', 'CongressmanHalRogers', 'SenatorMarcoRubio', 'jerrymoran', 'chrisvanhollen', 'RepDelBene', 'CongressmanJimHimes', 'RepRobWittman', 'CongressmanRickAllen', 'jiminhofe', 'RepEspaillat', 'RepDebbieDingell', 'RepBradWenstrup', 'RepKarenBass', 'TXRandy14', 'RepFredUpton', 'SenatorTedCruz', 'RepresentativeMarcyKaptur', 'RepMikeGallagher', 'RepDonYoung', '395759603917487', 'susancollins', 'RepBillJohnson', '118514606128', 'RepRonEstes', 'RepSeanMaloney', 'RepMcKinley', 'SenatorAngusSKingJr', 'senronjohnson', 'repdavidschweikert', 'RepMikeThompson', 'RepMoolenaar', 'RepAndyBiggs', 'RepStephMurphy', 'sanfordbishop', 'RepSarbanes', 'CongressmanGaryPalmer', 'RepSteveWomack', 'RepLeeZeldin', 'RepDanKildee', 'JoeWilson', '81125319109', 'congressmancomer', '96007744606', 'GusBilirakis', '115356957005', 'RepDwightEvans', 'RepPerlmutter', 'RepJackieWalorski', 'CongressmanBradSchneider', 'CongressmanFredKeller', 'aumuaamata', 'CongressmanJodyHice', 'reprichardneal', 'replahood', 'SenatorDurbin', 'RepGrothman', 'RepLouCorrea', 'SenatorWhitehouse', 'RepLaMalfa', 'CongressmanMcHenry', 'RepVirginiaFoxx', 'CongressmanGerryConnolly', 'SenDanSullivan', 'RepAndyBarr', 'repbrianfitzpatrick', 'mdiazbalart', 'SenatorBlunt', 'CongresswomanTitus', 'stevenpalazzo', 'USSenatorLindseyGraham', 'RepDrewFerguson', 'mikecrapo', 'SenatorStabenow', 'DutchRupp', 'lloyddoggett', 'CongressmanEricSwalwell', 'judgecarter', 'RepJimCosta', 'RepAnnaEshoo', 'RepJackyRosen', 'CongressmanRaulRuizMD', 'senatorhirono', 'RepThomasMassie', 'congresswomanbonamici', 'SenLisaMurkowski', 'RepWilson', 'RepNormaTorres', 'RepJohnKatko', 'RepGraceNapolitano', 'JohnBoozman', 'RepHakeemJeffries', 'RepJudyChu', 'RepDaveJoyce', 'MartinHeinrich', '191056827594903', 'reploudermilk', 'CongressmanMattGaetz', 'jefffortenberry', 'RepRickLarsen', 'SenBlumenthal', 'RepAnnWagner', 'michaeltmccaul', 'RepRodneyDavis', 'RichardShelby', 'RepChrisSmith', 'doris.matsui', 'RepRutherfordFL', 'repsaludcarbajal', 'PeterWelch', 'CongressmanAndreCarson', 'NancyPelosi', 'SenDuckworth', 'CongressmanGT', 'congressmanraja', 'CongresswomanNorton', 'senatormenendez', 'JimCooper', 'MaxineWaters', 'SteveDainesMT', 'USCongressmanVicenteGonzalez', 'RepCarolynMaloney', 'CongressmanDarrenSoto', 'senschumer', 'repstaceyplaskett', 'CongresswomanAdams', 'Rep.Billy.Long', 'congressmanbuddycarter', 'SenatorShaheen', 'USRepKathyCastor', 'boblatta', 'CongresswomanSinema', 'RepJennifferGonzalezColon', 'repbettymccollum', 'repyvettedclarke', 'RepCharlieCrist', 'usrepmikedoyle', 'senatortoomey', 'RepJayapal', '326420614138023', 'RepLowenthal', 'SenatorSherrodBrown', 'jerrymcnerney', 'repjasonsmith', 'CongresswomanEBJtx30', 'CongresswomanSheilaJacksonLee', '70063393423', '63158229861', 'RepWebster', '214258646163', 'JackieSpeier', 'kevinbrady', 'senrobportman', 'Rep.BluntRochester', 'RepBrianBabin', 'repraskin', 'reptomemmer', 'repdanbishop', 'repmikequigley', 'MarkRWarner', 'EdJMarkey', 'senatorsanders', 'repohalleran', 'grassley', 'RepRalphNorman', 'RepDWS', 'RepMorganGriffith', 'MarkAmodeiNV2', 'RepJimmyGomez', 'Rep.Grijalva', 'repronkind', 'janschakowsky', 'SenKirstenGillibrand', 'repkenbuck', 'reppeteaguilar', 'RepDavidEPrice', 'SenatorToddYoung', 'CongressmanJimLangevin', 'RepFrenchHill', 'CongresswomanClark', 'SenatorBobCasey', 'Congresswoman.Hartzler', 'derek.kilmer', 'reptrentkelly', 'SenMarkKelly', 'congressmanbobbyrush', 'CongresswomanBarragan', 'RepKayGranger', 'michaelcburgess', 'RepPeterDeFazio', 'SenatorWicker', 'RepMikeGarcia', 'RepAlLawsonJr', 'RepNewhouse', '8037068318'}

# set_of_pol_full_names = {'Brendan F. Boyle', 'Andy Kim', 'Lauren Underwood', 'Thomas P. Tiffany', 'Rand Paul', 'Mike Rogers', 'Salud O. Carbajal', 'Gary C. Peters', 'John Katko', 'Greg Pence', 'Eddie Bernice Johnson', 'Jim Costa', 'Aumua Amata Coleman Radewagen', 'Suzanne Bonamici', 'Tony Gonzales', 'Raja Krishnamoorthi', 'David B. McKinley', 'Stephanie I. Bice', 'Victoria Spartz', 'A. Drew Ferguson IV', 'Mary E. Miller', 'Lucille Roybal-Allard', 'Kirsten E. Gillibrand', 'Jeff Fortenberry', 'Darren Soto', 'John B. Larson', 'Troy Balderson', 'James Comer', 'Ken Buck', 'Russ Fulcher', 'Eric A. "Rick" Crawford', 'Brian Schatz', 'Doug Lamborn', 'Henry C. "Hank" Johnson, Jr.', 'Tom Emmer', 'Sam Graves', 'Alexandria Ocasio-Cortez', 'Richard Burr', 'Patty Murray', 'Ed Perlmutter', 'Charlie Crist', 'Kevin Brady', 'Lisa C. McClain', 'Cori Bush', 'Bill Cassidy', 'Mike Garcia', 'Ben Sasse', 'Tina Smith', 'Robert B. Aderholt', 'Mark DeSaulnier', 'Nicole Malliotakis', 'Al Green', 'Mariannette Miller-Meeks', 'Thomas R. Suozzi', 'Christopher H. Smith', 'Debbie Wasserman Schultz', 'Trey Hollingsworth', 'Mike Levin', 'Troy E. Nehls', 'Zoe Lofgren', 'Debbie Lesko', 'Mo Brooks', 'Matthew M. Rosendale', 'Mark E. Amodei', 'Rick W. Allen', 'Bill Hagerty', 'Donald Norcross', 'Marie Newman', 'Chris Van Hollen', 'Robert J. Wittman', 'Grace Meng', 'Nikema Williams', 'Dean Phillips', 'Pat Fallon', 'Eric Swalwell', 'Earl L. "Buddy" Carter', 'Kurt Schrader', 'Dan Bishop', 'Sherrod Brown', 'Ted Lieu', 'Jason Crow', 'Judy Chu', 'Cathy McMorris Rodgers', 'Anthony Gonzalez', 'Kaiali’i Kahele', 'Cynthia Axne', 'Jody B. Hice', 'Jack Bergman', 'Nancy Mace', 'Kevin Hern', 'Richard Blumenthal', 'Kathy E. Manning', 'David Kustoff', 'Greg Stanton', 'Richard C. Shelby', 'Michael T. McCaul', 'Tim Walberg', 'David N. Cicilline', 'Mark R. Warner', 'Paul Tonko', 'Val Butler Demings', 'Mark E. Green', 'Lois Frankel', 'Steve Daines', 'Tracey Mann', 'Guy Reschenthaler', 'Kat Cammack', 'Roy Blunt', 'André Carson', 'Alexander Mooney', 'Adrian Smith', 'Bob Gibbs', 'Pete Aguilar', 'Richard E. Neal', 'Ann Kirkpatrick', 'Fred Keller', 'Scott H. Peters', 'Andy Barr', 'Mark Kelly', 'Cindy Hyde-Smith', 'Joaquin Castro', 'Joe Wilson', 'Jack Reed', 'Lance Gooden', 'Thomas Massie', 'David Schweikert', 'Bill Johnson', 'John A. Yarmuth', 'Scott Fitzgerald', 'Roger Williams', 'Andy Biggs', 'Tom Malinowski', 'Joni Ernst', 'Angus S. King, Jr.', 'Patrick J. Leahy', 'C. A. Dutch Ruppersberger', 'John R. Moolenaar', 'Virginia Foxx', 'Tom Rice', 'John Thune', 'Barry Moore', 'Devin Nunes', 'Rashida Tlaib', 'Marc A. Veasey', 'Joe Neguse', 'Madison Cawthorn', 'Mike Bost', 'Pramila Jayapal', 'Bob Good', 'Jesús G. "Chuy" García', 'Larry Bucshon', 'Jared Huffman', 'Chris Jacobs', 'Mary Gay Scanlon', 'James E. Risch', 'Nancy Pelosi', 'Gregory W. Meeks', 'John W. Hickenlooper', 'Rick Scott', 'Jay Obernolte', 'Charles J. "Chuck" Fleischmann', 'Katherine M. Clark', 'Lizzie Fletcher', 'Jared F. Golden', 'Derek Kilmer', 'Tim Scott', 'Richard J. Durbin', 'Jim Cooper', 'Ann Wagner', 'Mondaire Jones', 'Michael C. Burgess', 'Cheri Bustos', 'August Pfluger', 'Sean Patrick Maloney', 'Edward J. Markey', 'Bernard Sanders', 'Ron Estes', 'Amy Klobuchar', 'Haley M. Stevens', 'Hakeem S. Jeffries', 'Dan Crenshaw', 'Tim Burchett', 'Tim Ryan', 'Michael K. Simpson', 'Diana DeGette', 'Todd Young', 'Melanie A. Stansbury', 'Lee M. Zeldin', 'Filemon Vela', 'Elissa Slotkin', 'Michael F. Bennet', 'Terri Sewell', 'Tom Cotton', 'Andrew R. Garbarino', 'Andrew S. Clyde', 'Markwayne Mullin', 'John Rose', 'James R. Langevin', 'Josh Harder', 'Burgess Owens', 'Stephen F. Lynch', 'Jake LaTurner', 'Rick Larsen', 'Jon Ossoff', 'Matt Cartwright', 'Andy Harris', 'Blaine Luetkemeyer', 'Michelle Steel', 'Garret Graves', 'Mikie Sherrill', 'Mike Quigley', 'Gregory F. Murphy', 'Richard Hudson', 'Liz Cheney', 'John P. Sarbanes', 'Elizabeth Warren', 'G. K. Butterfield', 'Ami Bera', 'Brad R. Wenstrup', 'James A. Himes', 'Neal P. Dunn', 'Byron Donalds', 'John Kennedy', 'Chellie Pingree', 'Chris Stewart', 'Bill Pascrell, Jr.', 'Brett Guthrie', 'Marsha Blackburn', 'Louie Gohmert', 'Lisa Blunt Rochester', 'Jenniffer González-Colón', 'Ken Calvert', 'Bradley Scott Schneider', 'Dan Sullivan', 'Don Bacon', 'Sean Casten', 'Brian K. Fitzpatrick', 'Kathleen M. Rice', 'Susie Lee', 'Yvette D. Clarke', 'Nydia M. Velázquez', 'Julia Brownley', 'Ayanna Pressley', 'William R. Timmons IV', 'Shelley Moore Capito', 'Mike Lee', 'Jahana Hayes', 'Michael F. Q. San Nicolas', 'Maria Elvira Salazar', 'Brian Babin', 'Chris Pappas', 'Carolyn Bourdeaux', 'Rob Portman', 'Maxine Waters', 'Jaime Herrera Beutler', 'Dianne Feinstein', 'Tom Cole', 'Anthony Brown', 'Steve Chabot', 'Ben Ray Luján', 'Thom Tillis', 'Mike Johnson', 'Karen Bass', 'Thomas R. Carper', 'Beth Van Duyne', 'Barry Loudermilk', 'Gus M. Bilirakis', 'Martin Heinrich', 'Donald S. Beyer, Jr.', 'Jennifer Wexton', 'Pete Sessions', 'Teresa Leger Fernandez', 'Gerald E. Connolly', 'Mike Kelly', 'Marco Rubio', 'Elaine G. Luria', 'Ted Budd', 'Rodney Davis', 'Adam Kinzinger', 'Yvette Herrell', 'Doug LaMalfa', 'Lucy McBath', 'Adam Smith', 'Lisa Murkowski', 'H. Morgan Griffith', 'Brenda L. Lawrence', 'Patrick T. McHenry', 'Jason Smith', 'Michael Guest', 'Ron Kind', 'Kyrsten Sinema', 'Jake Auchincloss', 'Bill Posey', 'Ralph Norman', 'Robert C. "Bobby" Scott', 'Michael Cloud', 'Jackie Speier', 'Margaret Wood Hassan', 'Tommy Tuberville', 'Gregorio Kilili Camacho Sablan', 'Dwight Evans', 'Ed Case', 'Josh Gottheimer', 'Lori Trahan', 'Lloyd Smucker', 'Raúl M. Grijalva', 'Jacky Rosen', 'Ron Wyden', 'Jimmy Gomez', 'David E. Price', 'J. Hill', 'Joe Manchin, III', 'Gwen Moore', 'Tom O’Halleran', 'Bobby L. Rush', 'Catherine Cortez Masto', 'Kim Schrier', 'Roger F. Wicker', 'Joseph D. Morelle', 'Dusty Johnson', 'Harold Rogers', 'Peter A. DeFazio', 'Trent Kelly', 'Christopher A. Coons', 'Lauren Boebert', 'Maria Cantwell', 'W. Gregory Steube', 'Patrick J. Toomey', 'Donald M. Payne, Jr.', 'Benjamin L. Cardin', 'Henry Cuellar', 'Steny H. Hoyer', 'Jim Banks', 'Josh Hawley', 'Carol D. Miller', 'Kathy Castor', 'Frederica S. Wilson', 'Robin L. Kelly', 'Tammy Baldwin', 'Paul A. Gosar', 'Elise M. Stefanik', 'James E. Clyburn', 'Robert P. Casey, Jr.', 'Daniel Webster', 'Cory A. Booker', 'Debbie Stabenow', 'Matt Gaetz', 'Steven Horsford', 'Ben Cline', 'James R. Baird', 'David G. Valadao', 'Lindsey Graham', 'Stephanie N. Murphy', 'Van Taylor', 'Vicente Gonzalez', 'Claudia Tenney', 'Don Young', 'David Rouzer', 'Peter Meijer', 'Nanette Diaz Barragán', 'Randy K. Weber, Sr.', 'Austin Scott', 'Tim Kaine', 'Tony Cárdenas', 'Joe Courtney', 'Tom Reed', 'Kay Granger', 'Ron Johnson', 'Warren Davidson', 'John H. Rutherford', 'Albio Sires', 'Ashley Hinson', 'Raphael G. Warnock', 'Fred Upton', 'John Barrasso', 'Ronny Jackson', 'Charles E. Schumer', 'Troy A. Carter', 'Seth Moulton', 'Tom McClintock', 'J. Luis Correa', 'Angie Craig', 'Frank D. Lucas', 'Roger Marshall', 'Mitch McConnell', 'Jackie Walorski', 'Joyce Beatty', 'Lloyd Doggett', 'David J. Trone', 'Brad Sherman', 'John R. Carter', 'Jamaal Bowman', 'Steve Womack', 'Jerry L. Carl', 'Michelle Fischbach', 'Mark Pocan', 'Glenn Grothman', 'Raul Ruiz', 'Sylvia R. Garcia', 'Jim Jordan', 'Mitt Romney', 'Deb Fischer', 'Peter Welch', 'Darrell Issa', 'James P. McGovern', 'Rosa L. DeLauro', 'Janice D. Schakowsky', 'Dina Titus', 'Sharice Davids', 'Mario Diaz-Balart', 'Blake D. Moore', 'Daniel T. Kildee', 'Jerry McNerney', 'Ritchie Torres', 'Adriano Espaillat', 'Andy Levin', 'Betty McCollum', 'Frank Pallone, Jr.', 'Randy Feenstra', 'John R. Curtis', 'Ann Kuster', 'Jim Hagedorn', 'Robert Menendez', 'Cynthia M. Lummis', 'Bennie G. Thompson', 'Jon Tester', 'Marcy Kaptur', 'Alma S. Adams', 'Suzan K. DelBene', 'Brian J. Mast', 'Jodey C. Arrington', 'James M. Inhofe', 'Conor Lamb', 'Madeleine Dean', 'Marjorie Taylor Greene', 'Steve Scalise', 'Sheila Jackson Lee', 'Daniel Meuser', 'Sanford D. Bishop, Jr.', 'Jeanne Shaheen', 'Ruben Gallego', 'A. Donald McEachin', 'Bill Huizenga', 'Doris O. Matsui', 'Bill Foster', 'Gary J. Palmer', 'Stacey E. Plaskett', 'Darin LaHood', 'Abigail Davis Spanberger', 'John Garamendi', 'Adam B. Schiff', 'David P. Joyce', 'Al Lawson, Jr.', 'Chrissy Houlahan', 'Mike Crapo', 'Glenn Thompson', 'Frank J. Mrvan', 'Jamie Raskin', 'Dan Newhouse', 'John Joyce', 'Susan Wild', 'Carlos A. Gimenez', 'Alex Padilla', 'Eleanor Holmes Norton', 'Alan S. Lowenthal', 'John Hoeven', 'Susan M. Collins', 'Mike Thompson', 'Pete Stauber', 'John Cornyn', 'Scott Perry', 'David Scott', 'Robert E. Latta', 'Jeff Duncan', 'Michael Waltz', 'Jimmy Panetta', 'Jefferson Van Drew', 'Veronica Escobar', 'Ilhan Omar', 'Anna G. Eshoo', 'Scott DesJarlais', 'Christopher Murphy', 'Chuck Grassley', 'William R. Keating', 'Debbie Dingell', 'Vern Buchanan', 'Chip Roy', 'Kelly Armstrong', 'Kevin McCarthy', 'Ro Khanna', 'Cliff Bentz', 'Diana Harshbarger', 'Brian Higgins', 'Norma J. Torres', 'Jerry Moran', 'Bryan Steil', 'Mike Braun', 'Carolyn B. Maloney', 'Earl Blumenauer', 'Deborah K. Ross', 'Mazie K. Hirono', 'Young Kim', 'Sara Jacobs', 'Barbara Lee', 'Tammy Duckworth', 'Julia Letlow', 'John Boozman', 'Bruce Westerman', 'Grace F. Napolitano', 'Linda T. Sánchez', 'Mark Takano', 'Katie Porter', 'Theodore E. Deutch', 'Mike Gallagher', 'Michael R. Turner', 'Ted Cruz', 'Sheldon Whitehouse', 'Bonnie Watson Coleman', 'James Lankford', 'Colin Z. Allred', 'Kevin Cramer', 'C. Scott Franklin', 'Steven M. Palazzo', 'Antonio Delgado', 'Jeff Merkley', 'Emanuel Cleaver', 'Danny K. Davis', 'Kweisi Mfume', 'Juan Vargas', 'Marilyn Strickland', 'Steve Cohen', 'Clay Higgins', 'Vicky Hartzler', 'Michael F. Doyle', 'Billy Long', 'Jerrold Nadler', 'Mike Rounds'}

# white_house = {'Joe Biden', 'Kamala Harris', 'Barack Obama', 'Donald J. Trump', 'Mike Pence'}

# combined = white_house.union(set_of_pol_full_names)

# #To check for substring
# data.loc[data['page_name'].str.contains('Congressman')]

# data.loc[data['page_name'].str.contains('Congresswoman')]

# data.loc[data['page_name'].str.contains('Senator')]
# #Results with this

# data.loc[data['page_name'].str.contains('Governor')]
# #Results with this

# #pol_fbpage_names = get_pol_fbpage_names()

# df_by_pol = data.loc[data['page_name'].isin(combined)]

# df_by_pol.groupby('page_name').count()